### PR TITLE
add GCS to cluster init script storage

### LINKED ISSUE #1203: [ISSUE] No GCS storage handler for clusters api InitScriptStorageInfo and
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Configuration
```hcl
  init_scripts {
    gs {
       destination="gs://databricks-shared/init_jars2.sh"
    }
  }
```

### Expected Behavior
GCS is an acceptable source for init scripts as per the databricks cluster ui on GCP: 
image

### Actual Behavior
 Error: Unsupported block type

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Terraform and provider versions

Terraform 1.1.7
Provider 0.5.2

### Debug Output
It's probably not needed given that there is no GCS storage option in the golang code.  
 
### Important Factoids

It still appears to be a missing format in the code at this point [here](https://github.com/databrickslabs/terraform-provider-databricks/blob/87709cd7c811d99eca15eefec03931d410918837/clusters/clusters_api.go#L201)  in InitScriptStorageInfo however databricks clusters does support GCS as an init script source, but it also appears undocumented:

https://docs.gcp.databricks.com/dev-tools/api/latest/clusters.html#clusterclusterinitscriptinfo

### PR DESCRIPTION
**Pull request by neilbest-db**
Intended to fix #1203.

Found no appropriate tests to mimic.

### CODE REVIEW

**Comment by alexott** [comment]
Please add documentation

File: go.mod [hunk 1]
```diff
@@ -14,6 +14,7 @@ require (
 	github.com/hashicorp/hcl v1.0.0
 	github.com/hashicorp/hcl/v2 v2.12.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.14.0
+	github.com/katbyte/terrafmt v0.4.0 // indirect
```
Comment (nfx) [inline go.mod:17]: Remove this. It doesn't belong to binary

**Review by nfx — CHANGES_REQUESTED** [review]
Small change required

File: go.mod (hunk 1 above)
Comment (nfx) [inline go.mod:17]:
```suggestion
```

**Review by nfx — APPROVED** [review]

**Pull request merged**